tests/features/async-components.spec.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  defineAsyncComponent,
+  defineComponent,
+  h,
+  ComponentPublicInstance
+} from 'vue'
+import flushPromises from 'flush-promises'
+
+import { mount } from '../../src'
+
+// AsyncComponents are documented here: https://github.com/vuejs/rfcs/blob/async-component/active-rfcs/0026-async-component-api.md
+
+describe('defineAsyncComponent', () => {
+  it('works with the basic usage', async () => {
+    const AsyncHello = defineAsyncComponent(() =>
+      import('../components/Hello.vue')
+    )
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(AsyncHello)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+    await flushPromises()
+    expect(wrapper.html()).toContain('Hello world')
+  })
+
+  it('works with options usage', async (done) => {
+    const Async = defineAsyncComponent({
+      loader: () =>
+        new Promise<any>((res) => {
+          setTimeout(() => {
+            res({
+              template: '<div>Async Component</div>'
+            })
+          }, 75)
+        }),
+      loadingComponent: {
+        template: '<div>Loading Component</div>'
+      },
+      delay: 10
+    })
+
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(Async)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+
+    setTimeout(() => {
+      expect(wrapper.html()).toContain('Loading Component')
+    }, 35)
+
+    setTimeout(() => {
+      expect(wrapper.html()).toContain('Async Component')
+      done()
+    }, 100)
+  })
+
+  it('works with vue files', async () => {
+    const Async = defineAsyncComponent({
+      loader: () => import('../components/Hello.vue')
+    })
+
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(Async)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+    await flushPromises()
+
+    expect(wrapper.html()).toContain('Hello world')
+  })
+
+  it('catches error and renders ErrorComponent', async () => {
+    const Async = defineAsyncComponent({
+      loader: () =>
+        new Promise<any>((res, rej) => {
+          rej('Async component failed to load.')
+        }),
+      errorComponent: {
+        template: '<div>Error Component</div>'
+      },
+      onError(error, retry, fail, attempts) {
+        fail()
+      }
```

Comment:
I think this `onError` is a bit redundant? The option is used to have some retry control, but here we're just failing no matter what.

Reply:
Hm, also the test is passing but it yields some warns and errors:

<img width="852" alt="image" src="https://user-images.githubusercontent.com/9197791/80873025-1277d080-8cb6-11ea-980f-a5132f3a91ca.png">

Reply:
Not sure if these error logs are somewhat expected, and we should just mock console for the test and call it a day… 🤔

Reply:
I had this thought too. Some other tests also throw and error. Is there any benefit to mocking the console? If printing the error is correct, I don't keep it.

Reply:
I tend to mute them to make it the output cleaner in the long run, but I don't care much either. Unless the console error is part of the expected output, then I spy it and assert the times called and so.

Reply:
(I still believe the `onError` bit can be removed)

Reply:
We can have our own Vue `errorHandler`, to easily assert on it without mocking `console.error`

Reply:
I will make this change, thanks